server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -325,6 +325,49 @@ class DeltaSharingService(serverConfig: ServerConfig) {
     streamingOutput(Some(queryResult.version), queryResult.responseFormat, queryResult.actions)
   }
 
+
+  @Post("/shares/{share}/schemas/{schema}/tables/{table}/queries/{queryId}")
+  @ConsumesJson
+  def getQueryStatus(
+     req: HttpRequest,
+     @Param("share") share: String,
+     @Param("schema") schema: String,
+     @Param("table") table: String,
+     @Param("queryId") queryId: String,
+     request: GetQueryInfoRequest): HttpResponse = processRequest {
+
+    if (table == "table1") {
+      throw new DeltaSharingIllegalArgumentException("expected error")
+    }
+
+    // we are reusing the table here to simulate a view query result
+    val tableConfig = sharedTableManager.getTable(share, schema, table)
+    val capabilitiesMap = getDeltaSharingCapabilitiesMap(
+      req.headers().get(DELTA_SHARING_CAPABILITIES_HEADER))
+    val responseFormatSet = getResponseFormatSet(capabilitiesMap)
+    val queryResult = deltaSharedTableLoader.loadTable(tableConfig).query(
+      includeFiles = true,
+      Seq.empty[String],
+      None,
+      None,
+      None,
+      None,
+      None,
+      None,
+      maxFiles = request.maxFiles,
+      pageToken = request.pageToken,
+      false,
+      None,
+      responseFormatSet = responseFormatSet)
+    if (queryResult.version < tableConfig.startVersion) {
+      throw new DeltaSharingIllegalArgumentException(
+        s"You can only query table data since version ${tableConfig.startVersion}."
+      )
+    }
+
+    streamingOutput(Some(queryResult.version), queryResult.responseFormat, queryResult.actions)
```

Comment:
Consider sending back a pending status with 50% likelihood. 
This would allow testing the wait loop on the client side.